api-stat.go

```diff
@@ -49,6 +49,30 @@ func (c Client) BucketExists(bucketName string) (bool, error) {
 	return true, nil
 }
 
+// List of headers to be ignored.
+var ignoreHeaders = []string{
+	"Transfer-Encoding",
+	"Accept-Ranges",
+	"Date",
+	"Server",
+	"Vary",
+	"x-amz-request-id",
+	"x-amz-id-2",
+	// Add new headers to be ignored.
+}
+
+// Cleanup all response headers which should be ignored
+// and removed from the response header map. Before
+// using them as metadata for the object.
+func cleanRespHeaders(headers http.Header, customIgnoreHeaders []string) {
+	for _, h := range customIgnoreHeaders {
+		headers.Del(h)
+	}
+	for _, h := range ignoreHeaders {
+		headers.Del(h)
+	}
+}
```

Comment:
better will be:
```
func extractMetadata(h http.Header) (metadata http.Header) {                                                                                                                      
        ignoreHeaders := map[string]bool {                                                                                                                                        
                "transfer-encoding": true,                                                                                                                                        
                "accept-ranges": true,                                                                                                                                            
                "date": true,                                                                                                                                                     
                ...                                                                                                                                                               
                ...                                                                                                                                                               
        }                                                                                                                                                                         
        for k, vv := range h {                                                                                                                                                    
                if !ignoreHeaders[k] {                                                                                                                                            
                        vv2 := make([]string, len(vv))                                                                                                                            
                        copy(vv2, vv)                                                                                                                                             
                        metadata[k] = vv2                                                                                                                                         
                }                                                                                                                                                                 
        }                                                                                                                                                                         
        return metadata                                                                                                                                                           
}
```

i.e the right function name and no mutation of resp.Header
also customIgnoreHeaders can be part of ignoreHeaders as it does not look like there will be any use for it keeping separate. If there is use for it in future then the function can be refactored.